### ISSUE TITLE
Custom includes not eager loading

### ISSUE THREAD

**Issue by Radiergummi** [spatie/laravel-query-builder#651]
TL;DR Custom includes doing the same thing as relationship includes don't eager load models, relationship includes do, even though the query itself is issued correctly.

-----

I have a situation where I'd like to limit an included resource to two fixed fields; API clients may not choose which fields to receive for this resource. At first, I tried combining the `AllowedInclude::relationship()` include with `allowedFields()` as suggested [in the documentation](https://spatie.be/docs/laravel-query-builder/v3/features/including-relationships#selecting-included-fields):
```php
$queryBuilder
    ->allowedFields('relation_table.col_1', 'relation_table.col_2')
    ->allowedIncludes('public_relation_identifier', 'actual_relation_name')
```

That yielded the following query, however:
```sql
select * from `relation_table` where `relation_table`.`model_id` in (42)
```
While the resulting models have properly hydrated relations, they contain all properties of the relation table, which includes heaps of data not required here.

So instead, I opted to create a [custom include class](https://spatie.be/docs/laravel-query-builder/v3/features/including-relationships#custom-includes) modelled after the [`IncludedRelationship` class](https://github.com/spatie/laravel-query-builder/blob/master/src/Includes/IncludedRelationship.php):
```php
class SparseInclude implements IncludeInterface
{
    public function __construct(private iterable $fields) {}

    public function __invoke(Builder $query, string $include)
    {
        $relatedTables = collect(explode('.', $include));
        $withs = $relatedTables
            ->mapWithKeys(function ($table, $key) use (
                $relatedTables
            ): array {
                $fullRelationName = $relatedTables
                    ->slice(0, $key + 1)
                    ->implode('.');

                if (empty($this->fields)) {
                    return [$fullRelationName];
                }

                return [
                    $fullRelationName => function ($query) {
                        $query->select($this->fields);
                    },
                ];
            })
            ->toArray();

        $query->with($withs);
    }
}

$stuff = QueryBuilder
    ::for(MyModel::class)
    ->allowedIncludes([
        AllowedInclude::custom('public_relation_identifier', new SparseInclude(['col_1', 'col_2']))
    ]);
```

This yields the (correct) relation query, which gives me lots of actual results if executed on the server:
```sql
select `col_1`, `col_2` from `relation_table` where `relation_table`.`model_id` in (42)
```

The API resource reports the relation is loaded (`$this->relationLoaded('actual_relation_name') === true`), but the relation collection is empty (`$this->actual_relation_name->count() === 0`).  
This means there seems to be a difference between how the relations are eager loaded between `AllowedInclude::custom()` and `AllowedInclude::relationship()` - but I can't figure out what it is...

**Comment by Radiergummi** [comment]
Just as usual, right after posting the issue I located the problem. I didn't care to include `model_id` with the query, which prevents Laravel from resolving the model correctly.
